src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -541,12 +561,15 @@ static ucs_status_t uct_tcp_sockcm_ep_recv_nb(uct_tcp_sockcm_ep_t *cep)
                                                              cep->comm_ctx.offset),
                                 &recv_length, NULL, NULL);
     if ((status != UCS_OK) && (status != UCS_ERR_NO_PROGRESS)) {
-        if (status == UCS_ERR_NOT_CONNECTED) {  /* ECONNRESET cannot return from recv() */
-            uct_tcp_sockcm_ep_handle_remote_disconnect(cep, status);
-        } else {
-            ucs_error("ep %p (fd=%d) failed to recv client's data (offset=%zu)",
-                      cep, cep->fd, cep->comm_ctx.offset);
+        if (status != UCS_ERR_NOT_CONNECTED) {  /* ECONNRESET cannot return from recv() */
+            ucs_error("ep %p (fd=%d) failed to recv client's data "
+                      "(offset=%zu status=%s)", cep, cep->fd, cep->comm_ctx.offset,
+                      ucs_status_string(status));
         }
+
+        /* treat all recv errors as if they are disconnect from the remote peer -
+         * i.e. stop sending and receiving on this endpoint */
+        uct_tcp_sockcm_ep_handle_remote_disconnect(cep, status);
```

Comment:
i would expect uct_tcp_sockcm_ep_send_skip_event() to not have any side effects, since it should only check for skip condition

Reply:
can you please explain what you mean and how uct_tcp_sockcm_ep_handle_remote_disconnect is related?

Reply:
uct_tcp_sockcm_ep_handle_remote_disconnect() has a side effect because it calls UCP callbacks
 uct_tcp_sockcm_ep_send_skip_event() should be just a predicate and not change any state

Reply:
where does uct_tcp_sockcm_ep_send_skip_event change the state? it only checks it.

Reply:
pls disregard, i thought the diff was coming from other function